source/plugin/Assets/GoogleMobileAds/Platforms/Unity/Countdown.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (C) 2020 Google LLC.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/** This class is responsible for the countdown timer shown on a rewarded ad.
+    After 5 seconds has elapsed, the close ad button will be set active and the user can then
+    close the ad. **/
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class Countdown : MonoBehaviour
+{
+
+    private float currentTime = 0f;
+    private float startingTime = 5f;
+    private Text[] countdownText;
```

Comment:
This class only needs access to 1 Text and 1 Button object. Don't need to have an array variable. Then button could have a better name, like closeButton?

Reply:
I definitely agree with naming the buttons but I need to have the countdownText equivalent to be an array since I'm getting components in children and there are 2 text objects in the prefab. I can name the array of Text components "texts" and Text[1] be countdownText

Reply:
So, in your Start() method, you can have code like:

```
Text[] texts = this.GetComponentsInChildren<Text>();
 if (texts.Length < 2)
 {
    Debug.Log("Invalid Prefab");
    return;
}
countdownText = texts[1];
```

And do the same for the buttons array. But you don't need to keep the entire array as class variables, you can let that local variable go out of scope now that you have the countdownText.

Reply:
So, this still isn't resolved. Reminder, instead of:

```
private Text[] texts;
private Button[] buttons;

public void Start() {
  buttons = ...
  texts = ....
}
```

you can write:

```
public void Start() {
  Button[] buttons = ...
  Text[] texts = ....
}

and get rid of the class variables.